### PR TITLE
initial code state

### PR DESCRIPTION
**Pull request by sn0wbi**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+FROM node:17.3.1 AS builder
```
Comment (patrick246) [inline Dockerfile:1]: 17.4.0 is already out

File: .env [hunk 2]
```diff
@@ -0,0 +1,15 @@
+BASIC_AUTH_USER='api-linter'
+BASIC_AUTH_PASSWORD='rD1cPU68xHOB9V'
```
Comment (patrick246) [inline .env]: leaked credentials?

File: README.md [hunk 3]
```diff
@@ -1,8 +1,74 @@
...
+
+## Running the API in "prod" mode
+
+```bash
+$ docker-compose -f dcp.yml up (-d)
+```
+
+### JS rules migration at application start up
+
+The API donwloads a dedicated spectral ruleset behind a configurable url as a base for further migration of this files into JavaScript. No need to redeploy the application after changing your spectral ruleset, restart is enough and you get served with your recently updated and published ruleset.
```
Comment (markbrockhoff) [inline README.md]:
```suggestion
The API downloads a dedicated spectral ruleset behind a configurable url as a base for further migration of this files into JavaScript. No need to redeploy the application after changing your spectral ruleset, restart is enough and you get served with your recently updated and published ruleset.
```

File: README.md [hunk 4]
```diff
@@ -1,8 +1,74 @@
-# api-linter-service
+# API Linting Service
 
-[![SIT](https://img.shields.io/badge/SIT-awesome-blueviolet.svg)](https://jobs.schwarz)
```
Comment (patrick246) [inline README.md:3]: You don't want to keep the SIT badge?

Comment (markbrockhoff) [inline README.md:3]: I think it would be better to keep the SIT Badge ;)

File: os_todo.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+Steps to migrate to OS @ Github
```
Comment (markbrockhoff) [inline os_todo.md]: Did you really want to commit this File?

File: README.md [hunk 6]
```diff
@@ -1,8 +1,74 @@
...
+
+## Spectral API Validation
+
+The API is build on top of the [@stoplight/spectral-core](https://www.npmjs.com/package/@stoplight/spectral-core) SDK for nodeJS to execute API validations that follow the "Open API Specification" pattern. Spectral itself is Open Source and able to validate any json or yaml defined schemas with dedicated rulesets, while playing it´s strengths definitely in validation of OpenAPI Specs.
+
+## Build / Setup the local dev env
+
+First make sure to have installed the following OS dependencies for development:
+* [Docker Desktop](https://www.docker.com/products/docker-desktop)
+* [node 17.3.1](https://nodejs.org/en/blog/release/v17.3.1/)
```
Comment (patrick246) [inline README.md]: Is this dependent on the exact version, or would aiming for the last or last LTS version be sufficient?

File: README.md (hunk 3 above)
Comment (patrick246) [inline README.md]: donwloads -> downloads

File: dcp.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+version: "3.7"
```
Comment (markbrockhoff) [inline dcp.yml:1]: There are to files at the moment: dcp.yml and docker-compose.yml they seem pretty redundant.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,97 @@
+{
+  "name": "api-linting-service",
+  "version": "1.0.0",
+  "description": "",
+  "author": "",
+  "private": true,
+  "license": "APACHE 2.0",
```
Comment (patrick246) [inline package.json]: NPM uses SPDX license codes in this field, for Apache 2.0 this should be `Apache-2.0`

**Review by markbrockhoff — CHANGES_REQUESTED** [review]

**Review by patrick246 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (sn0wbi) [reply README.md:3]: done

File: os_todo.md (hunk 5 above)
Comment (sn0wbi) [reply os_todo.md]: done

File: .env (hunk 2 above)
Comment (sn0wbi) [reply .env]: done

File: package.json (hunk 8 above)
Comment (sn0wbi) [reply package.json]: done

File: README.md (hunk 3 above)
Comment (sn0wbi) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (sn0wbi) [reply README.md]: LTS is fine

File: README.md (hunk 4 above)
Comment (sn0wbi) [reply README.md:3]: done

File: .env [hunk 9]
```diff
@@ -0,0 +1,12 @@
+SERVER_NAME_PROD='YOUR_PROD_SERVER_NAME'
+SERVER_NAME_LOCAL='http://localhost:3000'
+PORT=3001
```
Comment (markbrockhoff) [inline .env]: Is it intended that SERVER_NAME_LOCAL has a different port than PORT?

**Review by markbrockhoff — CHANGES_REQUESTED** [review]

File: .env (hunk 9 above)
Comment (sn0wbi) [reply .env]: done

**Review by markbrockhoff — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (thaberkern) [reply README.md:3]: I think he is removing it so I can do my traditional "add the SIT Badge" commit :)

**Review by patrick246 — APPROVED** [review]

**Pull request merged**